### ISSUE TITLE
[Question] Query returns an old DataSnapshot - and not new keys unless restarting the app

### ISSUE THREAD

**Issue by shlighter** [firebase/quickstart-unity#1146]
### [REQUIRED] Please fill in the following fields:

  Unity editor version: 2021.1.3f1
Firebase Unity SDK version:7.2.0
Source you installed the SDK: unitypackage (.unitypackage or Unity Package Manager)
Problematic Firebase Component: Database(Auth, Database, etc.)
Other Firebase Components in use: Database(Auth, Database, etc.)
Additional SDKs you are using: AdMob(Facebook, AdMob, etc.)
Platform you are using the Unity editor on: Windows(Mac, Windows, or Linux)
Platform you are targeting: Android(iOS, Android, and/or desktop)
Scripting Runtime: IL2CPP(Mono, and/or IL2CPP)

### [REQUIRED] Please describe the question here:
I retrieve 10 keys each time from my sorted DB.
`FirebaseDatabase.DefaultInstance.GetReference("CarsForSale/").OrderByKey().EndAt(lastSavedKey).LimitToLast(10).GetValueAsync()`

It works fine. But when new items are added to the database, I can't retrieve those new values unless I reset the app. If I don't use the query .OrderByKey().EndAt().LimitToLast, I do get the new keys from the DB without resetting the app
It's because query locally caches the values . I tried to use KeepSynced
But it only made my app very slow because that syncs the entire DB reference. I also tried to only use KeepSynced on the query of 5 keys but it doesn't make sense.
What can I do to make sure when I query keys from the DB to get latest the keys without having to restart my app or make it slow with coaching?

**Comment by paulinon** [comment]
Hi @shlighter,

I noticed that you're using an old version of the SDK. Could you confirm if this behavior exists in version 8.3.0 of the SDK? Could you also verify if this happens on both the editor and the Android build?

**Comment by shlighter** [comment]
Yes it happens with the new SDK as well and both editor and Android. Also I
haven't seen any new releases for this issue



[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality8&>
Sender
notified by Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality8&>
09/23/21,
09:40:03 AM

**Comment by paulinon** [comment]
Thanks for clarifying, @shlighter. In order to identify why this is happening, could you provide a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of your project along with the steps taken before facing the issue?

Could you also check if [adding event listeners](https://firebase.google.com/docs/database/unity/retrieve-data#listen_for_events) is applicable for your use case?
